src/model/networking/connection/tunnel/client-tunnel.cc

```diff
@@ -45,19 +41,19 @@ std::shared_ptr<Connection> ClientConnection::Create(
         return nullptr;
     }
 
-    std::shared_ptr<Connection> conn(new ClientConnection(
-        uuid,
+    // todo: refactor GenerateKeyPair and unique_ptr below to be more efficient
+    std::unique_ptr<Tunnel> tunnel(new ClientTunnel(
```

Comment:
I tried this before making the PR and kept getting compile errors. Its quite annoying because it should be fine with it. I will try it again to be sure.

Reply:
Also I think the fact that Tunnel has virtual functions means we cant use make on it anyway

Reply:
Went with a slightly different approach